Changelog.md

```diff
@@ -24,6 +24,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
     - `get_pipeline_executable_properties()`;
     - `get_pipeline_executable_statistics()`.
   The expected length of this array can be queried with the respective `*_len()` variant of these functions.
+- `push_next()` has been renamed to `extend()` and marked as `unsafe`. Users are encouraged to call `push()` for singular structs instead. (#909)
```

Comment:
Perhaps this should mention/describe that `push()` is new as well?

Either way, as with `0.38`, I'll write a migration guide that highlights a few features like this at the top of the README when releasing.

(Though could also write them now and submit it as part of this PR, let me know)